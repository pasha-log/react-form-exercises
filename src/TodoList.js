import React, {useState} from "react"; 
import {v4 as uuidv4} from "uuid";
import Todo from './Todo'
import NewTodoForm from "./NewTodoForm";

const TodoList = () => {
    const INITIAL_STATE = [
        {id: uuidv4(), todo: "Feed the cat"}, 
        {id: uuidv4(), todo: "Work on motorcycle"}
    ]
    const [todos, setTodos] = useState(INITIAL_STATE)
    const addTodo = (newTodo) => {
        setTodos(todos => [...todos, {...newTodo, id: uuidv4()}])
    }
    const deleteTodo = (id) => {
        setTodos(todos => 
            [...todos.filter(todo => todo.id !== id)]
        )
    }
    return (
        <div>
            <h3>Todo List</h3>
            <NewTodoForm addTodo={addTodo} />
            <div>
                {todos.map(({id, todo}) => <Todo id={id} key={id} todo={todo}  deleteTodo={deleteTodo}/>)}
            </div>

        </div>
    )
}

export default TodoList;

