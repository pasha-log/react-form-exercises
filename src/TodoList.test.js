import React from "react"; 
import {render, fireEvent} from "@testing-library/react";
import '@testing-library/jest-dom';
import TodoList from "./TodoList";

it("renders without crashing", function () {
    render(<TodoList />);
})

it("matches snapshot", function () {
    const {asFragment} = render(<TodoList />);
    expect(asFragment()).toMatchSnapshot();
})

it("should add new todo", function () {
    const {queryByText, getByLabelText} = render(<TodoList />)
    const input = getByLabelText("Add Todo");
    const btn = queryByText("Create Todo");
    expect(queryByText('Make breakfast')).not.toBeInTheDocument();
    fireEvent.change(input, {target: {value: 'Make breakfast'}})
    fireEvent.click(btn);
    expect(queryByText('Make breakfast')).toBeInTheDocument();
})