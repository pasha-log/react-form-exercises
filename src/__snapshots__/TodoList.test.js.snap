// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <h3>
      Todo List
    </h3>
    <form>
      <label
        for="todo"
      >
        Add Todo
      </label>
      <input
        id="todo"
        name="todo"
        placeholder="Make a Todo"
        type="text"
        value=""
      />
      <button>
        Create Todo
      </button>
    </form>
    <div>
      <ul>
        <li
          id="ae49d4f7-84cf-465f-91a7-f7c4a43ed091"
        >
          Feed the cat
          <button>
            X
          </button>
        </li>
      </ul>
      <ul>
        <li
          id="148dcbd3-14bf-444d-ade3-9be46f99b614"
        >
          Work on motorcycle
          <button>
            X
          </button>
        </li>
      </ul>
    </div>
  </div>
</DocumentFragment>
`;
